src/client/src/ui/analytics/views/analyticsBarChart.jsx

```diff
@@ -9,19 +10,29 @@ export default class PNLChart extends React.Component{
     isPnL: React.PropTypes.bool
   };
 
+  render(){
+    let bars = this._createBars();
+    return (
+      <div>{bars}</div>
+    );
+  }
+  
   _createBars(){
-
     const containerWidth = 330;
+
     let propName = this.props.isPnL ? CurrencyPairPosition.basePnlName : CurrencyPairPosition.baseTradedAmountName;
-    let maxMinValues = this.props.series.reduce( (resultObj, element) => {
+    let chartData = this.props.isPnL ? this.props.series : this._getPositionsDataFromSeries(propName);
+
+    let maxMinValues = chartData.reduce( (resultObj, element) => {
       resultObj.max = element[propName] > resultObj.max ? element[propName] : resultObj.max;
       resultObj.min = element[propName] < resultObj.min ? element[propName] : resultObj.min;
 
       return resultObj;
     }, {min: 0, max: 0});
 
     let ratio = (containerWidth/2) / Math.max(Math.abs(maxMinValues.max), Math.abs( maxMinValues.min));
-    let bars = this.props.series.map((element, index) => {
+    let bars = chartData.map((element, index) => {
+      if (element[propName] === 0 && !this.props.isPnL) return;
```

Comment:
so do we wan to `.filter()` instead of map? this just creates a sparse array with an undefined which react does not render but has to iterate through.